ledger/participant-integration-api/src/main/scala/platform/store/dao/events/TransactionIndexing.scala

```diff
@@ -77,6 +77,15 @@ object TransactionIndexing {
     private val disclosure = Map.newBuilder[NodeId, Set[Party]]
     private val visibility = Vector.newBuilder[(ContractId, Set[Party])]
 
+    init()
+
+    private def init(): Unit = {
+      for (contractId <- blinding.divulgence.keys) {
+        addDivulgence(contractId)
+      }
+      ()
+    }
```

Comment:
This removes the last (divulgence-related) dependency on transaction nodes to compute blinding info: it was previously driven by `Fetch` nodes (see below) and is now performed by iterating over divulgence information already present in `BlindingInfo`.

Without this change, some contracts are not correctly indexed.

Reply:
I noticed that you removed the divulgence coming from the blinding. Are you sure it will work? Are divulgences included in the divulgence list?

Reply:
Furthermore, if this is already broken on `master`, I am a bit concerned by the fact that we have not Ledger API test tool test that detects this.

Reply:
The divulgence addition I removed was driven by `Fetch` nodes in the transaction: the contract ID was taken from the `Fetch` node but used to look up the divulged parties in the blinding info.

Perhaps there was little point in letting `Fetch` nodes drive the divulgence additions even before this PR but it was still working; here it's not working anymore though, because `Fetch` nodes have been removed.

If blinding info computation is correct, then this behavior should be equivalent to the previous one. I don't know the extent of that coverage but I see there is some [here](https://github.com/digital-asset/daml/blob/master/daml-lf/engine/src/test/scala/com/digitalasset/daml/lf/engine/EngineTest.scala).

It looks to me like we have divulgence coverage as Ledger API test tool tests as well (e.g. [here](https://github.com/digital-asset/daml/blob/master/ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/tests/WitnessesIT.scala#L12)) and BTW divulgence-related tests were failing in this PR before the `TransactionIndexing` change (and the test I added was failing as well).

Which cases you think are not already covered?

Reply:
Don't mind the comments on this change, your replies in the other comment already address them. But thanks!